File: go/console/wizard/wizard_test.go
package wizard_test

import (
	"context"
	"errors"
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
	"hop.top/kit/go/console/wizard"
)

func TestNew_ValidSteps(t *testing.T) {
	w, err := wizard.New(
		wizard.TextInput("name", "Name"),
		wizard.Confirm("ok", "OK?"),
	)
	require.NoError(t, err)
	assert.NotNil(t, w)
}

func TestNew_DuplicateKey(t *testing.T) {
	_, err := wizard.New(
		wizard.TextInput("x", "A"),
		wizard.TextInput("x", "B"),
	)
	require.Error(t, err)
	assert.Contains(t, err.Error(), "duplicate")
}

func TestNew_BadDefaultType(t *testing.T) {
	_, err := wizard.New(
		wizard.Confirm("ok", "OK?").WithDefault("nope"),
	)
	require.Error(t, err)
	assert.Contains(t, err.Error(), "bool")
}

func TestNew_ActionWithoutFn(t *testing.T) {
	_, err := wizard.New(wizard.Step{
		Key: "act", Kind: wizard.KindAction, Label: "do",
	})
	require.Error(t, err)
	assert.Contains(t, err.Error(), "ActionFn")
}

func TestNew_SelectWithoutOptions(t *testing.T) {
	_, err := wizard.New(wizard.Step{
		Key: "sel", Kind: wizard.KindSelect, Label: "pick",
	})
	require.Error(t, err)
	assert.Contains(t, err.Error(), "options")
}

func TestAdvance_TextInput(t *testing.T) {
	w, err := wizard.New(wizard.TextInput("name", "Name"))
	require.NoError(t, err)

	res, err := w.Advance("Alice")
	require.NoError(t, err)
	assert.Nil(t, res)
	assert.Equal(t, "Alice", wizard.String(w.Results(), "name"))
	assert.True(t, w.Done())
}

func TestAdvance_Select(t *testing.T) {
	opts := []wizard.Option{
		{Value: "a", Label: "A"},
		{Value: "b", Label: "B"},
	}
	w, err := wizard.New(wizard.Select("color", "Color", opts))
	require.NoError(t, err)

	_, err = w.Advance("a")
	require.NoError(t, err)
	assert.Equal(t, "a", wizard.Choice(w.Results(), "color"))
}

func TestAdvance_Confirm(t *testing.T) {
	w, err := wizard.New(wizard.Confirm("ok", "Sure?"))
	require.NoError(t, err)

	_, err = w.Advance(true)
	require.NoError(t, err)
	assert.True(t, wizard.Bool(w.Results(), "ok"))
}

func TestAdvance_MultiSelect(t *testing.T) {
	opts := []wizard.Option{
		{Value: "x", Label: "X"},
		{Value: "y", Label: "Y"},
	}
	w, err := wizard.New(wizard.MultiSelect("tags", "Tags", opts))
	require.NoError(t, err)

	_, err = w.Advance([]string{"x", "y"})
	require.NoError(t, err)
	assert.Equal(t, []string{"x", "y"}, wizard.Strings(w.Results(), "tags"))
}

func TestAdvance_Action(t *testing.T) {
	called := false
	fn := func(_ context.Context, _ map[string]any) error {
		called = true
		return nil
	}
	w, err := wizard.New(wizard.Action("act", "Go", fn))
	require.NoError(t, err)

	res, err := w.Advance(nil)
	require.NoError(t, err)
	ar, ok := res.(*wizard.ActionRequest)
	require.True(t, ok)
	assert.Equal(t, "act", ar.StepKey)

	// Execute action and resolve.
	runErr := ar.Run(context.Background(), w.Results())
	require.NoError(t, w.ResolveAction(runErr))
	assert.True(t, called)
	assert.True(t, w.Done())
}

func TestAdvance_Validation(t *testing.T) {
	w, err := wizard.New(
		wizard.TextInput("email", "Email").
			WithValidateText(func(s string) error {
				if s == "bad" {
					return errors.New("invalid email")
				}
				return nil
			}),
	)
	require.NoError(t, err)

	_, err = w.Advance("bad")
	require.Error(t, err)
	var ve *wizard.ValidationError
	require.True(t, errors.As(err, &ve))
	assert.Equal(t, "email", ve.StepKey)
}

func TestAdvance_Required(t *testing.T) {
	w, err := wizard.New(
		wizard.TextInput("name", "Name").WithRequired(),
	)
	require.NoError(t, err)

	_, err = w.Advance("")
	require.Error(t, err)
	var ve *wizard.ValidationError
	require.True(t, errors.As(err, &ve))
	assert.Contains(t, ve.Error(), "required")
}

func TestResolveAction_Success(t *testing.T) {
	w, err := wizard.New(
		wizard.Action("a", "A", func(context.Context, map[string]any) error {
			return nil
		}),
		wizard.TextInput("b", "B"),
	)
	require.NoError(t, err)

	_, _ = w.Advance(nil) // get ActionRequest
	require.NoError(t, w.ResolveAction(nil))
	assert.Equal(t, "b", w.Current().Key)
}

func TestResolveAction_Abort(t *testing.T) {
	w, err := wizard.New(
		wizard.Action("a", "A", func(context.Context, map[string]any) error {
			return nil
		}).WithOnError(wizard.ActionAbort),
	)
	require.NoError(t, err)

	_, _ = w.Advance(nil)
	err = w.ResolveAction(errors.New("boom"))
	require.Error(t, err)
	var ae *wizard.ActionError
	require.True(t, errors.As(err, &ae))
	assert.Equal(t, wizard.ActionAbort, ae.Action)
}

func TestResolveAction_Retry(t *testing.T) {
	w, err := wizard.New(
		wizard.Action("a", "A", func(context.Context, map[string]any) error {
			return nil
		}).WithOnError(wizard.ActionRetry),
	)
	require.NoError(t, err)

	_, _ = w.Advance(nil)
	err = w.ResolveAction(errors.New("transient"))
	require.NoError(t, err)
	// Still on same step.
	assert.Equal(t, "a", w.Current().Key)
}

func TestResolveAction_Skip(t *testing.T) {
	w, err := wizard.New(
		wizard.Action("a", "A", func(context.Context, map[string]any) error {
			return nil
		}).WithOnError(wizard.ActionSkip),
		wizard.TextInput("b", "B"),
	)
	require.NoError(t, err)

	_, _ = w.Advance(nil)
	require.NoError(t, w.ResolveAction(errors.New("meh")))
	assert.Equal(t, "b", w.Current().Key)
}

func TestBack(t *testing.T) {
	w, err := wizard.New(
		wizard.TextInput("a", "A"),
		wizard.TextInput("b", "B"),
	)
	require.NoError(t, err)

	_, _ = w.Advance("hello")
	assert.Equal(t, "b", w.Current().Key)

	w.Back()
	assert.Equal(t, "a", w.Current().Key)
	// Result for "a" should be cleared.
	assert.Empty(t, wizard.String(w.Results(), "a"))
}

func TestBack_AtStart(t *testing.T) {
	w, err := wizard.New(wizard.TextInput("a", "A"))
	require.NoError(t, err)

	w.Back() // no-op
	assert.Equal(t, "a", w.Current().Key)
}

func TestConditional_When_True(t *testing.T) {
	w, err := wizard.New(
		wizard.Confirm("advanced", "Advanced?"),
		wizard.TextInput("extra", "Extra").
			WithWhen("advanced", func(v any) bool {
				b, _ := v.(bool)
				return b
			}),
	)
	require.NoError(t, err)

	_, _ = w.Advance(true)
	assert.Equal(t, "extra", w.Current().Key)
}

func TestConditional_When_False(t *testing.T) {
	w, err := wizard.New(
		wizard.Confirm("advanced", "Advanced?"),
		wizard.TextInput("extra", "Extra").
			WithWhen("advanced", func(v any) bool {
				b, _ := v.(bool)
				return b
			}),
	)
	require.NoError(t, err)

	_, _ = w.Advance(false)
	assert.True(t, w.Done())
}

func TestConditional_Back_OverSkipped(t *testing.T) {
	w, err := wizard.New(
		wizard.Confirm("show", "Show?"),
		wizard.TextInput("hidden", "Hidden").
			WithWhen("show", func(v any) bool {
				b, _ := v.(bool)
				return b
			}),
		wizard.TextInput("last", "Last"),
	)
	require.NoError(t, err)

	// show=false, hidden is skipped, land on "last".
	_, _ = w.Advance(false)
	assert.Equal(t, "last", w.Current().Key)

	// Back should skip "hidden" and land on "show".
	w.Back()
	assert.Equal(t, "show", w.Current().Key)
}

func TestDone(t *testing.T) {
	w, err := wizard.New(wizard.TextInput("a", "A"))
	require.NoError(t, err)

	assert.False(t, w.Done())
	_, _ = w.Advance("val")
	assert.True(t, w.Done())
}

func TestComplete_CallsOnComplete(t *testing.T) {
	w, err := wizard.New(wizard.TextInput("a", "A"))
	require.NoError(t, err)

	var got map[string]any
	w.SetOnComplete(func(r map[string]any) error {
		got = r
		return nil
	})

	_, _ = w.Advance("done")
	require.NoError(t, w.Complete())
	assert.Equal(t, "done", got["a"])
}

func TestComplete_DryRun(t *testing.T) {
	w, err := wizard.New(wizard.TextInput("a", "A"))
	require.NoError(t, err)

	called := false
	w.SetOnComplete(func(map[string]any) error {
		called = true
		return nil
	})
	w.SetDryRun(true)
	assert.True(t, w.DryRun())

	_, _ = w.Advance("val")
	require.NoError(t, w.Complete())
	assert.False(t, called)
}

func TestStepCount_ExcludesHidden(t *testing.T) {
	w, err := wizard.New(
		wizard.Confirm("show", "Show?"),
		wizard.TextInput("hidden", "Hidden").
			WithWhen("show", func(v any) bool {
				b, _ := v.(bool)
				return b
			}),
		wizard.TextInput("visible", "Visible"),
	)
	require.NoError(t, err)

	// Before answering, show defaults to nil → pred returns false.
	assert.Equal(t, 2, w.StepCount())

	_, _ = w.Advance(true)
	// Now hidden is visible.
	assert.Equal(t, 3, w.StepCount())
}

func TestNew_EmptyKey(t *testing.T) {
	_, err := wizard.New(wizard.Step{
		Key: "", Kind: wizard.KindTextInput, Label: "Name",
	})
	require.Error(t, err)
	assert.Contains(t, err.Error(), "key must not be empty")
}

func TestAdvance_Select_InvalidOption(t *testing.T) {
	opts := []wizard.Option{
		{Value: "a", Label: "A"},
		{Value: "b", Label: "B"},
	}
	w, err := wizard.New(wizard.Select("color", "Color", opts))
	require.NoError(t, err)

	_, err = w.Advance("z")
	require.Error(t, err)
	var ve *wizard.ValidationError
	require.True(t, errors.As(err, &ve))
	assert.Contains(t, ve.Error(), `invalid option "z"`)
}

func TestAdvance_MultiSelect_InvalidOption(t *testing.T) {
	opts := []wizard.Option{
		{Value: "x", Label: "X"},
		{Value: "y", Label: "Y"},
	}
	w, err := wizard.New(wizard.MultiSelect("tags", "Tags", opts))
	require.NoError(t, err)

	_, err = w.Advance([]string{"x", "nope"})
	require.Error(t, err)
	var ve *wizard.ValidationError
	require.True(t, errors.As(err, &ve))
	assert.Contains(t, ve.Error(), `invalid option "nope"`)
}

func TestCurrent_ClearsStaleResults(t *testing.T) {
	w, err := wizard.New(
		wizard.Confirm("show", "Show?"),
		wizard.TextInput("extra", "Extra").
			WithWhen("show", func(v any) bool {
				b, _ := v.(bool)
				return b
			}),
		wizard.TextInput("last", "Last"),
	)
	require.NoError(t, err)

	// Answer show=true, fill extra.
	_, _ = w.Advance(true)
	assert.Equal(t, "extra", w.Current().Key)
	_, _ = w.Advance("filled")
	assert.Equal(t, "filled", w.Results()["extra"])

	// Go back to "show", answer false this time.
	w.Back() // back to extra
	w.Back() // back to show
	_, _ = w.Advance(false)

	// "extra" is now skipped; its stale result must be cleared.
	assert.Equal(t, "last", w.Current().Key)
	assert.Empty(t, w.Results()["extra"])
}

func TestResultAccessors(t *testing.T) {
	results := map[string]any{
		"name":  "Alice",
		"ok":    true,
		"tags":  []string{"a", "b"},
		"color": "red",
		"bad":   42, // wrong type for all accessors
	}

	assert.Equal(t, "Alice", wizard.String(results, "name"))
	assert.Equal(t, "", wizard.String(results, "missing"))
	assert.Equal(t, "", wizard.String(results, "bad"))

	assert.True(t, wizard.Bool(results, "ok"))
	assert.False(t, wizard.Bool(results, "missing"))
	assert.False(t, wizard.Bool(results, "bad"))

	assert.Equal(t, []string{"a", "b"}, wizard.Strings(results, "tags"))
	assert.Nil(t, wizard.Strings(results, "missing"))
	assert.Nil(t, wizard.Strings(results, "bad"))

	assert.Equal(t, "red", wizard.Choice(results, "color"))
}

File: sdk/ts/src/wizard.test.ts
import { describe, it, expect } from "vitest";
import {
  StepKind,
  ErrorAction,
  Wizard,
  textInput,
  select,
  confirm,
  multiSelect,
  action,
  summary,
  ValidationError,
  ActionError,
  ActionRequest,
  resultString,
  resultBool,
  resultStrings,
  resultChoice,
  type Step,
} from "./wizard";

// --- construction ---

describe("Wizard construction", () => {
  it("accepts valid steps", () => {
    const w = new Wizard(
      textInput("name", "Name"),
      confirm("ok", "OK?"),
    );
    expect(w).toBeTruthy();
  });

  it("rejects duplicate keys", () => {
    expect(
      () => new Wizard(textInput("x", "A"), textInput("x", "B")),
    ).toThrow(/duplicate/i);
  });

  it("rejects empty key", () => {
    expect(
      () =>
        new Wizard({
          key: "",
          kind: StepKind.TextInput,
          label: "Name",
        } as Step),
    ).toThrow(/key must not be empty/i);
  });

  it("rejects bad default type for confirm", () => {
    expect(
      () => new Wizard(confirm("ok", "OK?").withDefault("nope")),
    ).toThrow(/bool/i);
  });

  it("rejects action without fn", () => {
    expect(
      () =>
        new Wizard({
          key: "act",
          kind: StepKind.Action,
          label: "do",
        } as Step),
    ).toThrow(/actionFn/i);
  });

  it("rejects select without options", () => {
    expect(
      () =>
        new Wizard({
          key: "sel",
          kind: StepKind.Select,
          label: "pick",
          options: [],
        } as Step),
    ).toThrow(/options/i);
  });

  it("auto-generates summary key", () => {
    const w = new Wizard(
      textInput("a", "A"),
      summary("Review"),
    );
    expect(w.current()!.key).toBe("a");
  });
});

// --- advance ---

describe("Advance", () => {
  it("text input stores string", () => {
    const w = new Wizard(textInput("name", "Name"));
    const [res, err] = w.advance("Alice");
    expect(err).toBeNull();
    expect(res).toBeNull();
    expect(resultString(w.results(), "name")).toBe("Alice");
    expect(w.done()).toBe(true);
  });

  it("select stores valid option", () => {
    const opts = [
      { value: "a", label: "A" },
      { value: "b", label: "B" },
    ];
    const w = new Wizard(select("color", "Color", opts));
    w.advance("a");
    expect(resultChoice(w.results(), "color")).toBe("a");
  });

  it("select rejects invalid option", () => {
    const opts = [
      { value: "a", label: "A" },
      { value: "b", label: "B" },
    ];
    const w = new Wizard(select("color", "Color", opts));
    const [, err] = w.advance("z");
    expect(err).toBeInstanceOf(ValidationError);
    expect(err!.message).toMatch(/invalid option/);
  });

  it("confirm stores bool", () => {
    const w = new Wizard(confirm("ok", "Sure?"));
    w.advance(true);
    expect(resultBool(w.results(), "ok")).toBe(true);
  });

  it("confirm rejects non-bool", () => {
    const w = new Wizard(confirm("ok", "Sure?"));
    const [, err] = w.advance("yes");
    expect(err).toBeInstanceOf(ValidationError);
  });

  it("multi_select stores string array", () => {
    const opts = [
      { value: "x", label: "X" },
      { value: "y", label: "Y" },
    ];
    const w = new Wizard(multiSelect("tags", "Tags", opts));
    w.advance(["x", "y"]);
    expect(resultStrings(w.results(), "tags")).toEqual(["x", "y"]);
  });

  it("multi_select rejects invalid option", () => {
    const opts = [
      { value: "x", label: "X" },
      { value: "y", label: "Y" },
    ];
    const w = new Wizard(multiSelect("tags", "Tags", opts));
    const [, err] = w.advance(["x", "nope"]);
    expect(err).toBeInstanceOf(ValidationError);
    expect(err!.message).toMatch(/invalid option.*nope/);
  });

  it("action returns ActionRequest", () => {
    let _called = false;
    const fn = async () => {
      _called = true;
    };
    const w = new Wizard(action("act", "Go", fn));
    const [res, err] = w.advance(null);
    expect(err).toBeNull();
    expect(res).toBeInstanceOf(ActionRequest);
    expect((res as ActionRequest).stepKey).toBe("act");
  });

  it("summary advances without value", () => {
    const w = new Wizard(
      textInput("a", "A"),
      summary("Review"),
    );
    w.advance("val");
    expect(w.current()!.kind).toBe(StepKind.Summary);
    w.advance(null);
    expect(w.done()).toBe(true);
  });

  it("validates required text", () => {
    const w = new Wizard(textInput("name", "Name").withRequired());
    const [, err] = w.advance("");
    expect(err).toBeInstanceOf(ValidationError);
    expect(err!.message).toMatch(/required/);
  });

  it("validates required multi_select", () => {
    const opts = [{ value: "x", label: "X" }];
    const w = new Wizard(
      multiSelect("tags", "Tags", opts).withRequired(),
    );
    const [, err] = w.advance([]);
    expect(err).toBeInstanceOf(ValidationError);
    expect(err!.message).toMatch(/required/);
  });

  it("runs custom text validator", () => {
    const w = new Wizard(
      textInput("email", "Email").withValidateText((s) => {
        if (s === "bad") throw new Error("invalid email");
      }),
    );
    const [, err] = w.advance("bad");
    expect(err).toBeInstanceOf(ValidationError);
    expect(err!.message).toMatch(/invalid email/);
  });
});

// --- resolveAction ---

describe("ResolveAction", () => {
  it("advances on success", () => {
    const w = new Wizard(
      action("a", "A", async () => {}),
      textInput("b", "B"),
    );
    w.advance(null); // get ActionRequest
    const err = w.resolveAction(null);
    expect(err).toBeNull();
    expect(w.current()!.key).toBe("b");
  });

  it("returns ActionError on abort", () => {
    const w = new Wizard(
      action("a", "A", async () => {}).withOnError(
        ErrorAction.Abort,
      ),
    );
    w.advance(null);
    const err = w.resolveAction(new Error("boom"));
    expect(err).toBeInstanceOf(ActionError);
    expect((err as ActionError).action).toBe(ErrorAction.Abort);
  });

  it("stays on step for retry", () => {
    const w = new Wizard(
      action("a", "A", async () => {}).withOnError(
        ErrorAction.Retry,
      ),
    );
    w.advance(null);
    const err = w.resolveAction(new Error("transient"));
    expect(err).toBeNull();
    expect(w.current()!.key).toBe("a");
  });

  it("skips on skip policy", () => {
    const w = new Wizard(
      action("a", "A", async () => {}).withOnError(
        ErrorAction.Skip,
      ),
      textInput("b", "B"),
    );
    w.advance(null);
    const err = w.resolveAction(new Error("meh"));
    expect(err).toBeNull();
    expect(w.current()!.key).toBe("b");
  });
});

// --- back ---

describe("Back", () => {
  it("goes to previous step and clears result", () => {
    const w = new Wizard(textInput("a", "A"), textInput("b", "B"));
    w.advance("hello");
    expect(w.current()!.key).toBe("b");
    w.back();
    expect(w.current()!.key).toBe("a");
    expect(resultString(w.results(), "a")).toBe("");
  });

  it("no-op at start", () => {
    const w = new Wizard(textInput("a", "A"));
    w.back();
    expect(w.current()!.key).toBe("a");
  });

  it("skips hidden steps during back", () => {
    const w = new Wizard(
      confirm("show", "Show?"),
      textInput("hidden", "Hidden").withWhen("show", (v) =>
        Boolean(v),
      ),
      textInput("last", "Last"),
    );
    w.advance(false);
    expect(w.current()!.key).toBe("last");
    w.back();
    expect(w.current()!.key).toBe("show");
  });
});

// --- conditional ---

describe("Conditional steps", () => {
  it("shows step when condition true", () => {
    const w = new Wizard(
      confirm("advanced", "Advanced?"),
      textInput("extra", "Extra").withWhen("advanced", (v) =>
        Boolean(v),
      ),
    );
    w.advance(true);
    expect(w.current()!.key).toBe("extra");
  });

  it("skips step when condition false", () => {
    const w = new Wizard(
      confirm("advanced", "Advanced?"),
      textInput("extra", "Extra").withWhen("advanced", (v) =>
        Boolean(v),
      ),
    );
    w.advance(false);
    expect(w.done()).toBe(true);
  });

  it("clears stale results on re-evaluation", () => {
    const w = new Wizard(
      confirm("show", "Show?"),
      textInput("extra", "Extra").withWhen("show", (v) =>
        Boolean(v),
      ),
      textInput("last", "Last"),
    );
    // show=true, fill extra
    w.advance(true);
    w.advance("filled");
    expect(w.results()["extra"]).toBe("filled");

    // go back, change show to false
    w.back(); // back to extra
    w.back(); // back to show
    w.advance(false);

    expect(w.current()!.key).toBe("last");
    expect(w.results()["extra"]).toBeUndefined();
  });
});

// --- done ---

describe("Done", () => {
  it("false initially, true after all steps", () => {
    const w = new Wizard(textInput("a", "A"));
    expect(w.done()).toBe(false);
    w.advance("val");
    expect(w.done()).toBe(true);
  });
});

// --- stepCount / stepIndex ---

describe("StepCount and StepIndex", () => {
  it("excludes hidden steps", () => {
    const w = new Wizard(
      confirm("show", "Show?"),
      textInput("hidden", "Hidden").withWhen("show", (v) =>
        Boolean(v),
      ),
      textInput("visible", "Visible"),
    );
    // before answering, show default nil → pred false
    expect(w.stepCount()).toBe(2);
    w.advance(true);
    expect(w.stepCount()).toBe(3);
  });

  it("stepIndex tracks visible position", () => {
    const w = new Wizard(
      textInput("a", "A"),
      textInput("b", "B"),
      textInput("c", "C"),
    );
    expect(w.stepIndex()).toBe(0);
    w.advance("x");
    expect(w.stepIndex()).toBe(1);
    w.advance("y");
    expect(w.stepIndex()).toBe(2);
  });
});

// --- dryRun / complete ---

describe("Complete and DryRun", () => {
  it("calls onComplete", () => {
    const w = new Wizard(textInput("a", "A"));
    let got: Record<string, unknown> | null = null;
    w.setOnComplete((r) => {
      got = r;
    });
    w.advance("done");
    w.complete();
    expect(got!["a"]).toBe("done");
  });

  it("skips onComplete in dryRun", () => {
    const w = new Wizard(textInput("a", "A"));
    let called = false;
    w.setOnComplete(() => {
      called = true;
    });
    w.setDryRun(true);
    expect(w.dryRun()).toBe(true);
    w.advance("val");
    w.complete();
    expect(called).toBe(false);
  });
});

// --- result accessors ---

describe("Result accessors", () => {
  const results: Record<string, unknown> = {
    name: "Alice",
    ok: true,
    tags: ["a", "b"],
    color: "red",
    bad: 42,
  };

  it("string", () => {
    expect(resultString(results, "name")).toBe("Alice");
    expect(resultString(results, "missing")).toBe("");
    expect(resultString(results, "bad")).toBe("");
  });

  it("bool", () => {
    expect(resultBool(results, "ok")).toBe(true);
    expect(resultBool(results, "missing")).toBe(false);
    expect(resultBool(results, "bad")).toBe(false);
  });

  it("strings", () => {
    expect(resultStrings(results, "tags")).toEqual(["a", "b"]);
    expect(resultStrings(results, "missing")).toBeNull();
    expect(resultStrings(results, "bad")).toBeNull();
  });

  it("choice (alias for string)", () => {
    expect(resultChoice(results, "color")).toBe("red");
  });
});
